### DISCUSSION TITLE
convergence wrt to plasmonics

### DISCUSSION THREAD

**Discussion by MahmutRuzi** [facebookresearch/fmmax#34] [Q&A]
Hello, dear developers,
First of all, thanks for the code. 

I am wondering if you have tested convergence on structures involved metals for plasmonics. I have tried a few RCWA & FMM codes, some do reasonably well (RETICOLO, for example) and some not so (an example is [torcwa](https://github.com/kch3782/torcwa).  Since you have implemented absorbing boundary layer, I guess in some situations it should work.  

For reference, I have a small [script](https://github.com/MahmutRuzi/photonics-sim-examples/blob/main/torcwa_AuNanobar2.ipynb) that uses torcwa. 

cheers

**Comment by mfschubert** [comment]
Hi @MahmutRuzi, thanks for your comment. The closest thing we have is the [metal grating](https://github.com/facebookresearch/fmmax/blob/main/examples/metal_grating.py) example, which models a 1D silver grating on a silver substrate, encapsulated with a n=1.5 dielectric.

Here is a quick plot made from the calculations performed by `metal_grating.convergence_study`. From this data, I would conclude that both TE- and TM-polarized reflection are relatively well converged for all formulations except `fft`.
![image](https://github.com/facebookresearch/fmmax/assets/30735893/883228de-d9b1-4de0-a3b1-aa1bb3b2b47d)


If you have a specific test that you think would be valuable to add (e.g. because it is comparable to published results elsewhere) we'd welcome the suggestion (or contribution).

**Reply by MahmutRuzi** [reply]
Hi @mfschubert, thanks for the quick response.  
Here is a test case that be considered. It is a 2D metallic crossing grating (L. Li, JOSAA 14, 2758-2767 (1997)) as shown Figure 8, 9, and 10 of the paper. The details were given in example 3. 

best

**Reply by mfschubert** [reply]
Sure thing. I created https://github.com/facebookresearch/fmmax/issues/37 calling for this example to be implemented. It would be a good first issue for contributors.
